### ISSUE TITLE
Missing Libs in release apk - Proguard Rules &  Android 11

### ISSUE THREAD

**Issue by gabidi** [Sifir-io/react-native-tor#14]
Some apps need to have this diff applied to their Proguard Rules to prevent it from over optimizing (striping) JNI code from this module.

https://github.com/Sifir-io/sifir-rs-sdk/commit/853ec84f49e09c8f7aed72e9731e22f56860e139

- If affected your build will crash on boot and catlog should show missing Libraries and/or JNI panics.
- The Proguard optimization step is often run only during release builds of most apps and so can go undetected while building and testing on debug builds.

Patched upstream, needs to be merged into next release.

**Comment by Overtorment** [comment]
Should this be reflected in the README..?

**Comment by Overtorment** [comment]
looks like the fix didn't help.

@godSaysHODL has this:

![image](https://user-images.githubusercontent.com/14336691/104458752-9c27af80-5560-11eb-8492-b96504aea1c3.png)

he is running grapheneOS (stripped android), so maybe its specific to this OS..?
@ncoelho tested it on a bunch of different android versions on BrowserStack and all seems to work, but those are stock androids.

**Comment by gabidi** [comment]
Noted and replied on PR.
Will get some time this weekend to look at it and update you.
Would love to get more users testing this on Graphene

**Comment by gabidi** [comment]
Turns out some behaviour changes are introduced in Anroid 10 and 11:
https://developer.android.com/about/versions/10/behavior-changes-all
https://developer.android.com/about/versions/10/behavior-changes-10

```
For compatibility, the symlinks are provided at the old paths. For example, /system/lib/libc.so is a symlink to /apex/com.android.runtime/lib/bionic/libc.so. So dlopen(“/system/lib/libc.so”) continues to work, but apps will find the difference when they actually try to examine the loaded libraries by reading /proc/self/maps or similar, which is not usual but we’ve found that some apps do that as part of their anti-hacking process. If so, the /apex/… paths should be added as valid paths for the Bionic files.
```

Patched System load in 
https://github.com/Sifir-io/react-native-tor/tree/fix/android11

Seems to be running well on Pixel 11 Browserstack.

**Comment by godSaysHODL** [comment]
> Patched System load in
> https://github.com/Sifir-io/react-native-tor/tree/fix/android11
> 
> Seems to be running well on Pixel 11 Browserstack.

You're a hero! Was worried I wasn't gonna get in on this tor fun on my pixel 3a💪

**Comment by gabidi** [comment]
Thanks @godSaysHODL Let's make sure it works :)

**Comment by gabidi** [comment]
Fixed by #17
